images/virtualization-artifact/pkg/common/network/network.go

```diff
@@ -47,17 +47,25 @@ type InterfaceStatus struct {
 
 type InterfaceSpecList []InterfaceSpec
 
-func CreateNetworkSpec(vmSpec virtv2.VirtualMachineSpec) InterfaceSpecList {
+func CreateNetworkSpec(vmSpec virtv2.VirtualMachineSpec, vmmacs []*virtv2.VirtualMachineMACAddress) InterfaceSpecList {
+	var macAddresses []string
+	for _, vmmac := range vmmacs {
+		macAddresses = append(macAddresses, vmmac.Status.Address)
+	}
```

Comment:
This array may have random order. It may cause a wrong assigning of macs to networks in the next loop.